File: codeabbey/059/MoraGender/codeabbey/059/Bulls and Cows.py
def calculate_hint(secret, guess):
    correct_position_count = sum(s == g for s, g in zip(secret, guess))
    correct_digit_count = sum(min(secret.count(digit), guess.count(digit)) for digit in set(guess))

    return f"{correct_position_count}-{correct_digit_count - correct_position_count}"


def main():
    input_data = input().split()
    secret = input_data[0]
    num_guesses = int(input_data[1])

    guesses = input().split()

    hints = [calculate_hint(secret, guess) for guess in guesses]

    print(" ".join(hints))


if __name__ == "__main__":
    main()

File: codeabbey/059/TanyaVon18/codeabbey/code/059/BullsandCows.java
import java.util.Scanner;

public class BullsandCows {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String[] input = scanner.nextLine().split(" ");
        String secret = input[0];
        

        String[] guesses = scanner.nextLine().split(" ");

        for (String guess : guesses) {
            String hint = calculateHint(secret, guess);
            System.out.print(hint + " ");
        }

        scanner.close();
    }

    public static String calculateHint(String secret, String guess) {
        int correctPositionCount = 0;
        int correctDigitCount = 0;

        for (int i = 0; i < secret.length(); i++) {
            if (secret.charAt(i) == guess.charAt(i)) {
                correctPositionCount++;
            } else if (secret.contains(String.valueOf(guess.charAt(i)))) {
                correctDigitCount++;
            }
        }

        return correctPositionCount + "-" + correctDigitCount;
    }
}
